clustermesh-apiserver/clustermesh/cells.go

```diff
@@ -46,13 +49,17 @@ var Cell = cell.Module(
 
 	k8sClient.Cell,
 	cmk8s.ResourcesCell,
-
 	kvstore.Cell,
-	cell.Provide(func(ss syncstate.SyncState) *kvstore.ExtraOptions {
+	dial.ClustermeshResolverCell,
+	cell.Provide(func(ss syncstate.SyncState, cmResolver *dial.ClustermeshResolver) *kvstore.ExtraOptions {
 		return &kvstore.ExtraOptions{
+			DialOption: []grpc.DialOption{
+				grpc.WithContextDialer(dial.NewContextDialer(log, cmResolver)),
+			},
 			BootstrapComplete: ss.WaitChannel(),
 		}
 	}),
+
```

Comment:
This container only accesses the sidecar etcd instance, hence the extra resolver logic is not necessary in this case.